staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go

```diff
@@ -151,6 +154,17 @@ func (o *APIResourceOptions) Complete(restClientGetter genericclioptions.RESTCli
 	o.groupChanged = cmd.Flags().Changed("api-group")
 	o.nsChanged = cmd.Flags().Changed("namespaced")
 
+	var printer printers.ResourcePrinter
+	if o.PrintFlags.OutputFormat != nil && len(*o.PrintFlags.OutputFormat) > 0 &&
+		(*o.PrintFlags.OutputFormat == "json" || *o.PrintFlags.OutputFormat == "yaml") {
+		printer, err = o.PrintFlags.ToPrinter()
```

Comment:
Based on what I wrote above it's sufficient to squash this condition to just:
```golang
if o.PrintFlags.OutputFormat != nil {
    printer, err = o.PrintFlags.ToPrinter()
    ...
```
since the rest will be handled by the printers, including the validation I wrote above.